### PR TITLE
[MRG] srFGW barycenters

### PR DESCRIPTION
**Pull request by cedricvincentcuaz**
## Types of changes

- add srFGW barycenters `semirelaxed_fgw_barycenters` with similar API than `fgw_barycenters`
- add generic function for update of barycenter structure and features in `gromov/_utils`: `update_barycenter_structure` and `update_barycenter_feature` which extend/relax several aspects of existing functions for gw barycenters (`update_square_loss`, `update_kl_loss` and `update_feature_matrix` --> **removed**), seeking for potential use in other variants than gw, e.g now used with sr(F)GW but could be also used with P(F)GW and U(F)GW.

## PR checklist

- [x] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [x] All tests passed, and additional code has been **covered with new tests**.
- [x] I have added the PR and Issue fix to the [**RELEASES.md**](RELEASES.md) file.

### CODE REVIEW

File: ot/gromov/_semirelaxed.py [hunk 1]
```diff
@@ -1100,3 +1105,244 @@ def entropic_semirelaxed_fused_gromov_wasserstein2(
...
+    d = Ys[0].shape[1]  # dimension on the node features
+
+    if fixed_structure:
+        if init_C is None:
+            raise UndefinedParameter('If C is fixed it must be initialized')
+        else:
+            C = init_C
+    else:
+        if init_C is None:
+            generator = check_random_state(random_state)
```
Comment (rflamary) [inline ot/gromov/_semirelaxed.py:1226]:
```suggestion
            rng = check_random_state(random_state)
```

File: ot/gromov/_semirelaxed.py [hunk 2]
```diff
@@ -1100,3 +1105,244 @@ def entropic_semirelaxed_fused_gromov_wasserstein2(
...
+            xalea = generator.randn(N, 2)
+            C = dist(xalea, xalea)
+            C = nx.from_numpy(C, type_as=ps[0])
+        else:
+            C = init_C
+
+    if fixed_features:
+        if init_X is None:
+            raise UndefinedParameter('If X is fixed it must be initialized')
```
Comment (rflamary) [inline ot/gromov/_semirelaxed.py:1235]:
```suggestion
            raise UndefinedParameter('If X is fixed it must be provided in init_X ')
```

File: ot/gromov/_semirelaxed.py [hunk 3]
```diff
@@ -1100,3 +1105,244 @@ def entropic_semirelaxed_fused_gromov_wasserstein2(
...
+    if log:
+        log_ = {}
+        if stop_criterion == 'barycenter':
+            log_['err_feature'] = []
+            log_['err_structure'] = []
+        else:
+            log_['loss'] = []
+            log_['err_rel_loss'] = []
+
+    while ((err_feature > tol or err_structure > tol or err_rel_loss > tol) and cpt < max_iter):
```
Comment (rflamary) [inline ot/gromov/_semirelaxed.py:1274]: easier to read when fo it in range(max_iter) and test with break au teh end of the loop

File: ot/gromov/_semirelaxed.py (hunk 1 above)
Comment (cedricvincentcuaz) [reply ot/gromov/_semirelaxed.py:1226]: done + same for other gw-like barycenter solvers.

File: ot/gromov/_semirelaxed.py (hunk 2 above)
Comment (cedricvincentcuaz) [reply ot/gromov/_semirelaxed.py:1235]: done + same for other gw-like barycenter solvers.

**Pull request merged**